apps/src/applab/applab.js

```diff
@@ -1486,11 +1486,12 @@ Applab.startInDesignMode = function () {
 };
 
 Applab.hideDesignModeToggle = function () {
-  return !!level.hideDesignMode;
+  return !!level.hideDesignMode || studioApp.share;
 };
 
 Applab.hideViewDataButton = function () {
-  return !!level.hideViewDataButton;
+  var isEditing = window.dashboard && window.dashboard.project.isEditing();
```

Comment:
we can't rely on appOptions.hideViewDataButton because it doesn't get updated when redirects between /view and /edit happen.